src/NServiceBus.RabbitMQ/Administration/DelayInfrastructure.cs

```diff
@@ -8,7 +8,9 @@
 
     static class DelayInfrastructure
     {
-        const int maxLevel = 32 - 1;
+        const int maxNumberOfBitsToUse = 28;
```

Comment:
We realized that we were encoding 32 bits/levels, but representing the delay as in `int` means we only have 31 bits available (because of the sign bit). We could either change the code to use a `uint` for the delay, or decrease the number of bits we're using.

We decided to decrease the number of bits, and while were were doing that, we thought it made sense to reevaluate how many we actually need. 28 bits gets us about 8 years worth of delay, so that seems like a reasonable upper bound.

However, we can always tweak this before merging, up to a max of 31 bits, or 32 if we make additional changes to accommodate a `uint`. We do have the idea of a negative delay, if we're told to delay into the past, so that would also have to be handled if we switched to `uint`.